### DISCUSSION TITLE
If there are many services

### DISCUSSION THREAD

**Discussion by Akenjeru** [rathole-org/rathole#360] [Q&A]
If I have a NAS on my home network, a conditional server on Ubuntu, proxmox, and so on. I just need to create on a server with a public IP additionally in server.toml the ones I need, for example, server.services.ubuntu, server.services.proxmox and configure it there as needed, right?
Or if I do this on my PROXMOX server where there are many different virtual machines and containers, will I not need to do it for each virtual machine separately or will I have to?

**Comment by typochecker** [comment]
各有优势。
如果只在局域网中配置一台rathole client，好处是你只需要维护一份client.toml
随之也有一些缺点，其他主机需要依赖这台client的宿主机。

在每一台需要暴露到公网的机器上配置rathole client，是一个虽然有点费力，但是能做到解耦的方式。如果是我，我就在每台机器上都配置client，因为它占用的资源很少。

**Reply by Akenjeru** [reply]
I have a personal server and there is Portainer and there are many different containers inside, will I need to install the client only on Portainer so that I can access other different containers?
and what is the correct way to configure client.toml in this case? Everyone will have the same IP address, but different ports
